### PR TITLE
more flexibility for config, add std out for sshd

### PR DESCRIPTION
**Pull request by bubenkoff**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ method = scp
 login = reprepro
 allow_unsigned_uploads = 0
 allowed_distributions = wheezy
-post_upload_command = ssh %(login)s@%(fqdn)s reprepro processincoming incoming
+post_upload_command = ssh %(login)s@%(fqdn)s reprepro include /data/debian/incoming
```
Comment (bbinet) [inline README.md]: why this change?

File: Dockerfile [hunk 2]
```diff
@@ -9,16 +9,20 @@ RUN mkdir /var/run/sshd
...
 
 ADD sshd_config /sshd_config
 ADD run.sh /run.sh
 RUN chmod +x /run.sh
 
 ENV REPREPRO_DEFAULT_NAME Reprepro
+ENV REPREPRO_ALLOW wheezy>wheezy/dev
+ENV REPREPRO_CODENAME wheezy
+ENV REPREPRO_CODENAME_DEV wheezy/dev
+ENV REPREPRO_CODENAME_PROD wheezy/prod
```
Comment (bbinet) [inline Dockerfile]: I'm not sure about this change because it's still quite limited (for example we still cannot use any number of distributions/architectures we want) and if one is not happy with the default configuration, he can already use its customized reprepro configuration using a docker volume.
I would accept a pull request that would allow more exhaustive configuration though (be able to use one or many custom distributions/architectures...)

File: Dockerfile [hunk 3]
```diff
@@ -9,16 +9,20 @@ RUN mkdir /var/run/sshd
 RUN echo "REPREPRO_BASE_DIR=/data/debian" > /etc/environment
 
 # Configure an reprepro user (admin)
-RUN adduser --system --group --shell /bin/bash --uid 600 --disabled-password --no-create-home reprepro
+RUN adduser --system --group --shell /bin/bash --uid 600 --disabled-password --home=/data reprepro
 
 # Configure an apt user (read only)
-RUN adduser --system --group --shell /bin/bash --uid 601 --disabled-password --no-create-home apt
+RUN adduser --system --group --shell /bin/bash --uid 601 --disabled-password --home=/data apt
```
Comment (bbinet) [inline Dockerfile]: why do you want to create a home directory? and then who will own `/data` dir, apt user or reprepro user?

File: run.sh [hunk 4]
```diff
@@ -44,7 +44,7 @@ else
     echo "=> /data/debian directory does not exist:"
     echo "   Configuring a default debian repository with reprepro..."
 
-    keyid=$(gpg --dry-run /config/reprepro_pub.gpg | sed "s/.*\/\([^ ]*\).*/\1/")
+    keyid=$(gpg --dry-run /config/reprepro_pub.gpg | grep pub | sed "s/.*\/\([^ ]*\).*/\1/")
```
Comment (bbinet) [inline run.sh]: do we really need this additional grep?

File: run.sh [hunk 5]
```diff
@@ -63,34 +63,33 @@ EOF
...
 EOF
-    touch /data/debian/conf/override.wheezy
+    touch /data/debian/conf/override.${REPREPRO_CODENAME}
     chown -R reprepro:reprepro /data/debian
 fi
 
 echo "=> Starting SSH server..."
-exec /usr/sbin/sshd -f /sshd_config -D
-
+exec /usr/sbin/sshd -f /sshd_config -D -e
```
Comment (bbinet) [inline run.sh]: good idea, thanks.

File: run.sh (hunk 4 above)
Comment (bubenkoff) [inline run.sh]: depending on the version of gpg, it outputs more additional text, didn't work for me without this grep

File: Dockerfile (hunk 3 above)
Comment (bubenkoff) [inline Dockerfile]: that helped me because i was able to log in by ssh to check what's accessible, i agree it's strange to share home folder anyway

Comment (bbinet) [inline Dockerfile]: you should be able to log in anyway

File: run.sh (hunk 4 above)
Comment (bbinet) [inline run.sh]: Ok, thanks for your explanation

**Comment by bbinet** [comment]
See 0f3b218 for a more flexible approach.
Closing this PR, now that the master branch provides all features from this PR.

**Comment by bubenkoff** [comment]
thanks a lot for doing this!
I'll play with it

**Comment by bubenkoff** [comment]
checked the latest version and it works very well (except small warning about the home directories, see issue #4)
thank you very much!
